src/gt4sd/algorithms/generation/diffusion/geodiff/model/core.py

```diff
@@ -0,0 +1,335 @@
+#
+# MIT License
+#
+# Copyright (c) 2022 GT4SD team
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+from typing import Tuple, Union
+
+import torch
+from diffusers.configuration_utils import ConfigMixin
+from diffusers.modeling_utils import ModelMixin
+from torch import nn
+from torch_geometric.data import Data
+
+from .layers import (
+    GINEncoder,
+    MLPEdgeEncoder,
+    MoleculeGNNOutput,
+    MultiLayerPerceptron,
+    SchNetEncoder,
+)
+from .utils import (
+    assemble_atom_pair_feature,
+    clip_norm,
+    extend_graph_order_radius,
+    get_distance,
+    graph_field_network,
+    is_local_edge,
+)
+
+
+class MoleculeGNN(ModelMixin, ConfigMixin):
+    """Graph Neural Network Model for molecule conformation."""
+
+    def __init__(
+        self,
+        hidden_dim: int = 128,
+        num_convs: int = 6,
+        num_convs_local: int = 4,
+        cutoff: float = 10.0,
+        mlp_act: str = "relu",
+        edge_order: int = 3,
+        edge_encoder: str = "mlp",
+        smooth_conv: bool = True,
+    ):
+        """
+        Args:
+            hidden_dim: Hidden dimension of the GNN.
+            num_convs: Number of GNN layers.
+            num_convs_local: Number of GNN layers for local edges.
+            cutoff: Cutoff radius for the GNN.
+            mlp_act: Activation function for the MLP.
+            edge_order: Order of the edge features.
+            edge_encoder: Type of edge encoder.
+            smooth_conv: Whether to use smooth convolution.
+        """
```

Comment:
```suggestion
        """Constructs a MoleculeGNN.

        Args:
            hidden_dim: hidden dimension of the GNN. Defaults to 120.
            num_convs: number of GNN layers. Defaults to 6.
            num_convs_local: number of GNN layers for local edges. Defaults to 4.
            cutoff: cutoff radius for the GNN. Defaults to 10.0.
            mlp_act: activation function for the MLP. Defaults to "relu".
            edge_order: order of the edge features. Defaults to 3.
            edge_encoder: type of edge encoder. Defaults to "mlp".
            smooth_conv: whether to use smooth convolution. Defaults to True.
        """
```
In case the implementation was adapted from a source I would reiterate it also in this docstring.